packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts

```diff
@@ -306,4 +332,22 @@ export class Schedule extends Resource implements ISchedule {
     const isEmptyPolicy = Object.values(policy).every(value => value === undefined);
     return !isEmptyPolicy ? policy : undefined;
   }
-}
\ No newline at end of file
+
+  private validateTimeFrame(startDate?: string, endDate?: string) {
+    if (startDate && !Schedule.ISO8601_REGEX.test(startDate)) {
+      throw new Error(`startDate needs to follow the format yyyy-MM-ddTHH:mm:ss.SSSZ but got ${startDate}`);
+    }
+    if (endDate && !Schedule.ISO8601_REGEX.test(endDate)) {
+      throw new Error(`endDate needs to follow the format yyyy-MM-ddTHH:mm:ss.SSSZ but got ${endDate}`);
+    }
+
+    if (startDate && endDate) {
+      const start = new Date(startDate);
+      const end = new Date(endDate);
+
+      if (end <= start) {
+        throw new Error(`startDate must come before the endDate but got startDate: ${startDate}, endDate: ${endDate}`);
+      }
+    }
```

Comment:
```suggestion
    if (startDate && endDate && startDate > endDate) {
      throw new Error(`startDate must preceed endDate, got startDate: ${startDate}, endDate: ${endDate}`);
    }
```
It should be fine to compare the strings directly given the format.